lib/autoproj/ops/configuration.rb

```diff
@@ -227,17 +242,30 @@ def load_and_update_package_sets(root_pkg_set, only_local = false)
                 end
                 by_repository_id[repository_id] = [vcs, imported_from]
 
+                # Make sure the package set has been already checked out to
+                # retrieve the actual name of the package set
                 if !vcs.local?
                     update_remote_package_set(vcs, only_local)
-                    create_remote_set_user_dir(vcs)
+                    raw_local_dir = PackageSet.raw_local_dir_of(vcs)
+                    required_remotes_dirs << raw_local_dir
                 end
-
                 name = PackageSet.name_of(manifest, vcs)
+
+                names = by_name.collect { |k,v| k }
+                Autoproj.debug "Trying to load package_set: #{name} from definition #{repository_id}"
+                Autoproj.debug "Already loaded package_sets are: #{names}"
+
                 if already_loaded = by_name[name]
                     already_loaded_pkg_set, already_loaded_vcs = *already_loaded
                     if already_loaded_vcs != vcs
                         if imported_from
-                            Autoproj.warn "#{imported_from.name} auto-imports a package set from #{vcs}, but a package set with the same name (#{name}) has already been imported from #{already_loaded_vcs}, I am skipping this one"
+                            Autoproj.warn "redundant auto-import by #{imported_from.name} for package set '#{name}'."
+                            Autoproj.warn "    A package set with the same name (#{name}) has already been imported from"
+                            Autoproj.warn "        #{already_loaded_vcs}"
+                            Autoproj.warn "        with server configuration '#{repository_configuration_label(already_loaded_vcs)}'."
```

Comment:
I am trying to make the error message readable to a user!
Since the user only does:
 -- Autoproj.gitorious_server_configuration('GITORIOUS', 'gitorious.org')
Rendering it into a url e.g. repository_id=gitorious://... is an autoproj interna.